src/main.rs

```diff
@@ -41,82 +50,61 @@ fn main() {
         .arg(
             Arg::with_name("net")
                 .long("net")
-                .help("Network parameters \"tap=<if_name>,ip=<ip_addr>,mask=<net_mask>,mac=<mac_addr>\"")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name("cpus")
-                .long("cpus")
-                .help("Number of virtual CPUs")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name("memory")
-                .long("memory")
-                .help("Amount of RAM (in MB)")
+                .help(
+                    "Network parameters \"tap=<if_name>,\
+                     ip=<ip_addr>,mask=<net_mask>,mac=<mac_addr>\"",
+                )
                 .takes_value(true),
         )
         .arg(
             Arg::with_name("rng")
                 .long("rng")
                 .help("Path to entropy source")
-                .default_value("/dev/urandom"),
+                .default_value(config::DEFAULT_RNG_SOURCE),
         )
         .get_matches();
 
-    let kernel_arg = cmd_arguments
+    let cpus = cmd_arguments.value_of("cpus");
+
+    let memory = cmd_arguments.value_of("memory");
+
+    let kernel = cmd_arguments
         .value_of("kernel")
-        .map(PathBuf::from)
         .expect("Missing argument: kernel");
-    let kernel_path = kernel_arg.as_path();
 
-    let disk_paths: Vec<PathBuf> = cmd_arguments
+    let cmdline = cmd_arguments.value_of("cmdline");
+
+    let disk: Vec<&str> = cmd_arguments
         .values_of("disk")
-        .expect("Missing arguments on disk")
-        .map(PathBuf::from)
+        .expect("Missing argument: disk. Provide at least one")
         .collect();
 
-    let cmdline = cmd_arguments
-        .value_of("cmdline")
-        .map(std::string::ToString::to_string)
-        .unwrap_or_else(String::new);
-
-    let net_params = cmd_arguments
-        .value_of("net")
-        .map(std::string::ToString::to_string);
+    let net = cmd_arguments.value_of("net");
 
-    let rng_path = match cmd_arguments.occurrences_of("rng") {
-        0 => None,
-        _ => Some(cmd_arguments.value_of("rng").unwrap().to_string()),
-    };
+    let rng = cmd_arguments.value_of("rng");
```

Comment:
FYI you changed the meaning of RNG meaning it is no longer optional. From the docs:

>NOTE: If the user does not use this argument at runtime ArgMatches::is_present will still return true. If you wish to determine whether the argument was used at runtime or not, consider ArgMatches::occurrences_of which will return 0 if the argument was not used at runtime.

Reply:
We discussed this was fine offline.